qcodes/instrument_drivers/stanford_research/SR86x.py

```diff
@@ -156,17 +158,53 @@ def snapshot_base(self, update: bool = False,
         snapshot = super().snapshot_base(update, params_to_skip_update)
         return snapshot
 
-    @staticmethod
-    def _set_capture_len_parser(value: int) -> int:
+    def _set_capture_len_parser(self, capture_len_in_kb: int) -> int:
+        """
+        According to the manual, the capture length is set in kB. This method checks for the range of the input value,
+        and throws an error if the range is not satisfied. Additionally, if the input value is odd, it is made even
+        (and a warning is issued) since the capture length can only be set in even numbers.
 
-        if value % 2:
-            log.warning("the capture length needs to be even. Setting to {}".format(value + 1))
-            value += 1
+        Args:
+            capture_len_in_kb (int): The desired capture length in kB.
 
-        if not 1 <= value <= 4096:
-            raise ValueError("the capture length should be between 1 and 4096")
+        Returns:
+            capture_len_in_kb (int)
+        """
+        return self._parse_capture_length(capture_len_in_kb, issue_warning=True)
+
+    def _parse_capture_length(self, capture_length_in_kb: int, issue_warning: bool=False) -> int:
+        """
+        Parse the capture length in kB according to the way buffer treats it. This method checks for the range
+        of the input value, and throws an error if the range is not satisfied. Additionally, if the input value
+        is odd, it is made even (and a warning is issued, if desired) since the capture length can only be set
+        in even numbers.
+
+        This method intended for use in other methods inside this class for calculation purposes. Setting issue_warning
+        argument to True allows this method to be used as well for set commands of the instrument parameters.
+
+        Args:
+            capture_length_in_kb (int): The desired capture length in kB.
+            issue_warning (bool): If True, a warning is issued when setting capture length to an odd number
+        Returns:
+            capture_len_in_kb (int)
+        """
+        if capture_length_in_kb % 2:
+            capture_length_in_kb += 1
```

Comment:
I don't think it is a good idea to give the user anything else than either exactly what they ask for or a big fat exception. I realise that this is a private method, but I think that principle is still valid.